docs/ibc/1_IBC_ARCHITECTURE.md

```diff
@@ -0,0 +1,154 @@
+
+### Authentication
+
+All datagrams in IBC are authenticated: a block finalized by the consensus algorithm of the sending ledger must commit to the outgoing datagram via a cryptographic accumulator, and the receiving chain's IBC handler must verify both the consensus transcript and the cryptographic accumulator proof that the datagram was sent (and associated actions executed) before acting upon it.
+
+### Statefulness
+
+Reliability, flow control, and authentication as described above require that IBC initializes and maintains certain status information for each datastream. This information is called a connection. Each connection object contains information about the consensus state of the connected ledger, negotiated encoding & multiplexing options, and state & sequence numbers. When two modules wish to communicate, they must locate an existing connection between their two ledgers, or initialize a new connection if none yet exists. Initializing a connection requires a multi-step handshake which, once complete, ensures that only the two intended ledgers are connected and future datagrams relayed will be authenticated, encoded, and sequenced as desired.
+
+### Multiplexing
+
+To allow for many modules within a single host ledger to use an IBC connection simultaneously, IBC provides a set of channels within each connection, which each uniquely identify a linear datapipe over which packets can be sent in order to a destination module on the receiving ledger. Channels are usually expected to be associated with a single module on each ledger, but one-to-many and many-to-one channels are also possible. The number of channels is unbounded, facilitating concurrent throughput limited only by the throughput of the underyling ledgers with only a single connection necessary to track consensus information (and consensus transcript verification cost thus amortized across all channels using the connection).
+
+## Dataflow
+
+IBC can be conceptualized as a layered protocol stack, through which data flows top-to-bottom (when sending IBC packets) and bottom-to-top (when receiving IBC packets).
+
+The "handler" is the part of the state machine implementing the IBC protocol, which is responsible for translating calls from modules to and from packets and routing them appropriately to and from channels & connections.
+
+Consider the path of an IBC packet between two chains — call them *A* and *B*:
+
+### Diagram
+
+```
++------------------------------------------------------------------------------+
+| Chain A                                                                      |
+|                                                                              |
+| Module --> Handler --> Packet --> Channel --> Connection --> Consensus       |
++------------------------------------------------------------------------------+
```

Comment:
I think this deserves slightly more structure. In combination with the diagram under "Protocol relations" (which gives a really good high level overview IMO), this is slightly confusing: If you look at both diagrams together, it looks like "Consensus" here is part of the IBC box (under "Protocol relations"). @cwgoes clarified to me (thanks!) that this is the actual Consensus though (as in tendermint Consensus and not the Consensus state as per ICS2).

Reply:
Added more structure.